src/sentry/integrations/utils/code_mapping.py

```diff
@@ -285,6 +283,27 @@ def _potential_match(self, src_file: str, frame_filename: FrameFilename) -> bool
             match = (
                 split[0].rfind(f"/{frame_filename.root}") > -1 or split[0] == frame_filename.root
             )
+        return match
+
+    def _potential_match_no_transformation(
+        self, src_file: str, frame_filename: FrameFilename
+    ) -> bool:
+        # static/app/utils/handleXhrErrorResponse.tsx && app/utils/handleXhrErrorResponse.tsx
+        return src_file.rfind(frame_filename.full_path) > -1
+
+    def _potential_match(self, src_file: str, frame_filename: FrameFilename) -> bool:
+        """Tries to see if the stacktrace without the root matches the file from the
+        source code. Use existing code mappings to exclude some source files
+        """
+        # Exit early because we should not be processing source files for existing code maps
+        if self._matches_current_code_mappings(src_file, frame_filename):
+            return False
+
+        match = False
+        if frame_filename.full_path.endswith(".py"):
+            match = self._potential_match_with_transformation(src_file, frame_filename)
+        else:
+            match = self._potential_match_no_transformation(src_file, frame_filename)
```

Comment:
No preference, but this could also be 

```suggestion
        if frame_filename.full_path.endswith(".py"):
            return self._potential_match_with_transformation(src_file, frame_filename)
        return self._potential_match_no_transformation(src_file, frame_filename)
```

Reply:
Yes both are possible but I follow this specific design pattern where returns are only at the end of a function. In larger functions it is a noticeably better pattern.